feedback-post-admin-options.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * The admin page for Feedback Post plugin.
+ *
+ * @package feedback_post_plugin
+ */
+
+ /*
+ Copyright 2017 OCAD University
+
+ Licensed under the Educational Community License (ECL), Version 2.0 or the New
+ BSD license. You may not use this file except in compliance with one these
+ Licenses.
+
+ You may obtain a copy of the ECL 2.0 License and BSD License at
+ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+ */
+
+add_action ('admin_menu', 'feedback_post_plugin_menu');
+add_action ('admin_init', 'feedback_post_plugin_menu_init');
+
+function feedback_post_plugin_menu() {
+    add_options_page (
+        'Feedback Post Plugin Options',
+        'Feedback Post Plugin Options',
+        'manage_options',
+        'feedback-post-plugin',
+        'create_feedback_post_plugin_options_page'
+    );
+}
+
+/*
+Specify what fields appear in the admin page.
+*/
+function feedback_post_plugin_menu_init() {
+    register_setting (
+        'feedback-post-plugin-settings-group',
+        'feedback-post-plugin-settings',
+        'feedback_post_plugin_settings_validate'
+    );
+
+    // Section containing the selector for choosing the feedback post success page.
+    add_settings_section (
+        'feedback-post-plugin-settings-section',
+        'Settings for the Feedback Post Plugin',
+        'feedback_post_plugin_settings_section_callback',
+        'feedback-post-plugin'
+    );
+
+    // The selector for choosing the feedback post submission form success page.
+    add_settings_field (
+        'success-id',
+        'Submit Success Page',
+        'feedback_post_plugin_success_id_callback',
+        'feedback-post-plugin',
+        'feedback-post-plugin-settings-section'
+    );
+}
+
+/*
+Callback function for the settings section. Currently this is empty as there
+is only 1 setting on the page and is fairly self descriptive.
+ */
+function feedback_post_plugin_settings_section_callback() {
```

Comment:
Do we need this function that does nothing?

Reply:
That callback is required by [add_settings_section](https://codex.wordpress.org/Function_Reference/add_settings_section).

Instead we can pass it an anonymous empty function:

```
add_settings_section (
        'feedback-post-plugin-settings-section',
        'Settings for the Feedback Post Plugin',
        function() {},
        'feedback-post-plugin'
    );
```
Would that be better?
```

Reply:
I just tried replacing the callback with an anonymous function and it causes the page selector to not appear. So it would seem it's needed.

Looks like we will need to keep it, even though it is empty @cindyli.